extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js

```diff
@@ -59,15 +59,24 @@ export class AmpAdNetworkFakeImpl extends AmpA4A {
           // and the signature is "FAKESIG". This mode is only allowed in
           // `localDev` and primarily used for A4A Envelope for testing.
           // See DEVELOPING.md for more info.
-          const creative = this.transformCreativeLocalDev_(deserialized);
+          let creative = this.transformCreativeLocalDev_(deserialized);
+          try {
+            // Gracefully handle the case in which the fake ad is still
+            // formatted in JSON, which it almost certainly is, since
+            // "gulp serve" won't serve an HTML file under /extensions
+            const json = JSON.parse(deserialized);
```

Comment:
Sorry, should have made my question clearer. And this is mainly for my own learning. Let me know if any of my understanding went wrong:

- In fast fetch, runtime first requests a URL via XHR. It returns a JSON (AMP creative) or HTML (non-AMP creative)
- In case of non-AMP or invalid AMP, runtime creates an iframe and having the `src=` attribute pointing to the same URL (the cache frame rendering path). And we expect it to hit the browser cache to save a round trip.
- In case of non-AMP, the HTML is perfectly rendered in the iframe. However, for the invalid AMP case, how do we deal with the cached JSON?

@keithwrightbos